### PR TITLE
Rework of func class: Reformatting, PHPDoc for methods, removed dead code, removed unused methods

### PR DESCRIPTION
**Pull request by andygrunwald**
- No functionality change in methods
- Reformatting for proper readability
- Added proper PHPDocs for methods
- Removed dead code
- Removed unused methods

### CODE REVIEW

File: inc/classes/class_func.php [hunk 1]
```diff
@@ -200,67 +195,56 @@ public function unixstamp2date($func_timestamp, $func_art)
...
-          return $func_date;
-    }
 
-  /**
-   * Calculates the age to a given birthday timestamp
-   * @timestamp birthhday
-   * @optional timestamp attime Calculates the age the person will have at this time
-   * @return int age
-   */
-    public function age($birthday, $attime = 0)
```
Comment (M4LuZ) [inline inc/classes/class_func.php]: I'm wondering:
* Why a user-age related function is in `func` and not in `user`
* Why this is not used anywhere, although there has to be functionality to evaluate the minimum age. Duplicated code?
* Why [DateTime::diff](http://php.net/manual/de/datetime.diff.php) isn't used to provide this. Guess this is pre-PHP-5.3 developed code.

I agree with the removal from `func`, but would suggest to have a look at the current age verification system, if a similar function in user exists (or should be created)

File: inc/classes/class_func.php [hunk 2]
```diff
@@ -396,7 +449,6 @@ public function text2html($string, $mode = 0)
             }
 
             if ($mode != 2) {
-                $img_start = "<img src=\"design/".$auth["design"]."/images/";
```
Comment (M4LuZ) [inline inc/classes/class_func.php]: Why was this removed?

File: inc/classes/class_func.php [hunk 3]
```diff
@@ -561,21 +621,10 @@ public function Entity2Uml($string)
         return $string;
     }
 
-
-  #### Check Var Content ####
-    public function check_var($var, $type, $min_length, $max_length)
```
Comment (M4LuZ) [inline inc/classes/class_func.php]: Fine to remove. 
There is native PHP support with the `is_<type>()` functions, and the error message does not really provide added value, does it? ;)

File: inc/classes/class_func.php (hunk 2 above)
Comment (andygrunwald) [reply inc/classes/class_func.php]: Because the variable `$img_start` is not in use at all.
In this line, it gets written, but no one (in the complete file) is accessing/reading this variable.
See https://github.com/lansuite/lansuite/blob/master/inc/classes/class_func.php#L399

File: inc/classes/class_func.php [hunk 4]
```diff
@@ -594,99 +644,127 @@ public function checkIP($ip)
...
+     * @param int       $max_entries_per_page
+     * @param int       $overall_entries
+     * @param string    $working_link
+     * @param string    $var_page_name
+     * @return array
+     */
     public function page_split($current_page, $max_entries_per_page, $overall_entries, $working_link, $var_page_name)
     {
         if ($max_entries_per_page > 0 and $overall_entries >= 0 and $working_link != "" and $var_page_name != "") {
-            if ($current_page == "") {
```
Comment (M4LuZ) [inline inc/classes/class_func.php]: Reason for removal?

File: inc/classes/class_func.php (hunk 3 above)
Comment (andygrunwald) [reply inc/classes/class_func.php]: Exactly. Once we are getting more and more stable we can introduce PHP native type system here.

File: inc/classes/class_func.php (hunk 4 above)
Comment (andygrunwald) [reply inc/classes/class_func.php]: Here it will be checked if `$current_page` is empty. If empty, then the vars `$page_sql`, `$page_a` and `$page_b` will be set.
A couple of lines after `$current_page` will be checked again and has an `else` part.
In this `else` part the variables `$page_sql`, `$page_a` and `$page_b` will be overwritten.

See https://github.com/lansuite/lansuite/blob/master/inc/classes/class_func.php#L638-L651

As a result: This is dead code.

File: inc/classes/class_func.php (hunk 2 above)
Comment (M4LuZ) [reply inc/classes/class_func.php]: Ah, understood!

File: inc/classes/class_func.php (hunk 1 above)
Comment (M4LuZ) [reply inc/classes/class_func.php]: There is already an option to define a minimum age for a party and AFAIK this is also enforced on signup, so similar code must already exist at another place. But I would also need to search for it first.

But as said not blocking the removal here, may just be worth a look.

File: inc/classes/class_func.php (hunk 4 above)
Comment (M4LuZ) [reply inc/classes/class_func.php]: Indeed, good find!
